core/src/Cuda/Kokkos_Cuda_Team.hpp

```diff
@@ -262,13 +263,18 @@ class CudaTeamMember {
   template <typename ReducerType>
   KOKKOS_INLINE_FUNCTION static std::enable_if_t<is_reducer<ReducerType>::value>
   vector_reduce(ReducerType const& reducer) {
-    vector_reduce(reducer, reducer.reference());
+    using value_type = typename ReducerType::value_type;
+    using wrapped_reducer_type =
+        typename Impl::FunctorAnalysis<Impl::FunctorPatternInterface::REDUCE,
+                                       TeamPolicy<Cuda>, ReducerType,
+                                       value_type>::Reducer;
+
+    vector_reduce(wrapped_reducer_type(reducer), reducer.reference());
   }
 
-  template <typename ReducerType>
+  template <typename ReducerType, typename ValueType>
   KOKKOS_INLINE_FUNCTION static std::enable_if_t<is_reducer<ReducerType>::value>
-  vector_reduce(ReducerType const& reducer,
-                typename ReducerType::value_type& value) {
+  vector_reduce(ReducerType const& reducer, ValueType& value) {
```

Comment:
Same here. SInce this is user-facing we shouldn't expect a wrapped reducer to be passed in but wrap the reducer in this function instead.

Reply:
Changed so that existing function signatures remain unchanged. Instead introduced a function that will take in a wrapped reducer.